data-model/src/main/scala/za/co/absa/enceladus/model/properties/PropertyDefinition.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.model.properties
+
+import java.time.ZonedDateTime
+
+import com.fasterxml.jackson.databind.node.{ArrayNode, ObjectNode}
+import za.co.absa.enceladus.model.menas.MenasReference
+import za.co.absa.enceladus.model.menas.audit.{AuditFieldName, AuditTrailChange, AuditTrailEntry, Auditable}
+import za.co.absa.enceladus.model.properties.essentiality.{Essentiality, Mandatory, Optional}
+import za.co.absa.enceladus.model.properties.propertyType.PropertyType
+import za.co.absa.enceladus.model.versionedModel.VersionedModel
+
+case class PropertyDefinition(name: String,
+                              version: Int = 1,
+                              description: Option[String] = None,
+
+                              propertyType: PropertyType,
+                              putIntoInfoFile: Boolean = false,
+                              essentiality: Essentiality = Optional(),
+                              disabled: Boolean = false,
```

Comment:
Wasn't disabled included in `Essentiality`?

Reply:
@benedeki `disabled` field is enforced by the `VersionedModel`, so it would require quite a bit of reengineering to get rid of that. Having it present, we inherit the option to disable by an API call etc.

@AdrianOlosutean what do you mean? However, if we could simplify the `essentiality` type on that account (e.g. to a boolean field), I'd be glad.